src/util/subscribeToResult.ts

```diff
@@ -68,15 +67,18 @@ export function subscribeToResult<T>(outerSubscriber: OuterSubscriber<any, any>,
         break;
       }
     } while (true);
-  } else if (typeof result[$$observable] === 'function') {
+  } else if (result && typeof result[$$observable] === 'function') {
     const obs = result[$$observable]();
     if (typeof obs.subscribe !== 'function') {
-      destination.error(new Error('invalid observable'));
+      destination.error(new TypeError('Provided object does not correctly implement Symbol.observable'));
     } else {
       return obs.subscribe(new InnerSubscriber(outerSubscriber, outerValue, outerIndex));
     }
   } else {
-    destination.error(new TypeError('unknown type returned'));
+    const value = isObject(result) ? 'an invalid object' : `'${result}'`;
+    const msg = `You provided ${value} where a stream was expected.`
+      + ' You can provide an Observable, Promise, Array, or Iterable.';
```

Comment:
if you use template strings, the indention will matter and be wack. I would need to do this:

```js
    const msg = `You provided ${value} where a stream was expected.
You can provide an Observable, Promise, Array, or Iterable.`;
// Notice how this can't be indented, otherwise the indention will also transfer through
// e.g. this:
    const msg = `You provided ${value} where a stream was expected.
      You can provide an Observable, Promise, Array, or Iterable.`;
// would output:
You provided null where a stream was expected.
      You can provide an Observable, Promise, Array, or Iterable.
````

If you're OK with that, I can change it. lmk.